raui-core/src/animator.rs

```diff
@@ -24,49 +36,154 @@ impl AnimationUpdate {
         Self { states, update }
     }
 
+    /// Check whether or not the widget has an animation with the given `anim_id`
     #[inline]
-    pub fn has(&self, id: &str) -> bool {
-        self.states.has(id)
+    pub fn has(&self, anim_id: &str) -> bool {
+        self.states.has(anim_id)
     }
 
+    /// Change the animation associated to a given `anim_id`
     #[inline]
-    pub fn change(&self, name: &str, data: Option<Animation>) -> Result<(), AnimationError> {
-        self.update.change(name, data)
+    pub fn change(
+        &self,
+        anim_id: &str,
+        animation: Option<Animation>,
+    ) -> Result<(), AnimationError> {
+        self.update.change(anim_id, animation)
     }
 
-    /// (progress factor, time, duration)
+    /// Get the current progress of the animation of a given value
+    ///
+    /// This will return [`None`] if the value is not currently being animated.
     #[inline]
-    pub fn value(&self, id: &str, name: &str) -> Option<(Scalar, Scalar, Scalar)> {
-        self.states.value(id, name)
+    pub fn value_progress(&self, anim_id: &str, value_name: &str) -> Option<AnimatedValueProgress> {
+        self.states.value_progress(anim_id, value_name)
     }
 
+    /// Get the current progress factor of the animation of a given value
+    ///
+    /// If the value is currently being animated this will return [`Some`] [`Scalar`] between `0`
+    /// and `1` with `0` meaning just started and `1` meaning finished.
+    ///
+    /// If the value is **not** currently being animated [`None`] will be returned
     #[inline]
-    pub fn value_progress(&self, id: &str, name: &str) -> Option<Scalar> {
-        self.value(id, name).map(|v| v.0)
+    pub fn value_progress_factor(&self, anim_id: &str, value_name: &str) -> Option<Scalar> {
+        self.states
+            .value_progress(anim_id, value_name)
+            .map(|x| x.progress_factor)
     }
 
+    /// Same as [`value_progress_factor`][Self::value_progress_factor] but returning `default` instead of [`None`]
     #[inline]
-    pub fn value_progress_or(&self, id: &str, name: &str, v: Scalar) -> Scalar {
-        self.value_progress(id, name).unwrap_or(v)
+    pub fn value_progress_factor_or(
+        &self,
+        anim_id: &str,
+        value_name: &str,
+        default: Scalar,
+    ) -> Scalar {
+        self.value_progress_factor(anim_id, value_name)
+            .unwrap_or(default)
     }
 
+    /// Same as [`value_progress_factor`][Self::value_progress_factor] but returning `0` instead of [`None`]
     #[inline]
-    pub fn value_progress_or_zero(&self, id: &str, name: &str) -> Scalar {
-        self.value_progress(id, name).unwrap_or(0.0)
+    pub fn value_progress_factor_or_zero(&self, anim_id: &str, value_name: &str) -> Scalar {
+        self.value_progress_factor(anim_id, value_name)
+            .unwrap_or(0.)
     }
 }
 
+/// The amount of progress made for a value in an animation
```

Comment:
add derive for Debug, Default, Clone (and maybe Copy since this isn't big struct?) - this type is purely user-side and user might want to debug and process the data and functionalities of these derives are mostly used on the user-side processing.